Lux meter crashes when closing it

**Actual Behaviour**

When using Lux meter and pressing back, it crashes

**Expected Behaviour**

Application should open instruments page when pressed back from Lux meter instead of crashing

**Steps to reproduce it**

1. Open app
2. Go to `Lux Meter`
3. Go to `Configure` tab
4. Change `High Limit` text box value
5. Click back or move to other tab and observe app is crashing

**LogCat for the issue**

```
FATAL EXCEPTION: main
Process: org.fossasia.pslab, PID: 31407
java.lang.RuntimeException: Unable to destroy activity {org.fossasia.pslab/org.fossasia.pslab.activity.LuxMeterActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'android.text.Editable android.widget.EditText.getText()' on a null object reference
at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:3769)
at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:3787)
at android.app.ActivityThread.access$1400(ActivityThread.java:149)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1378)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:211)
at android.app.ActivityThread.main(ActivityThread.java:5321)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1016)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.text.Editable android.widget.EditText.getText()' on a null object reference
at org.fossasia.pslab.fragment.LuxMeterFragmentConfig$4.onFocusChange(LuxMeterFragmentConfig.java:131)
at android.view.View.onFocusChanged(View.java:5258)
at android.widget.TextView.onFocusChanged(TextView.java:7940)
at android.view.View.clearFocusInternal(View.java:5141)
at android.view.View.unFocus(View.java:5174)
at android.view.ViewGroup.unFocus(ViewGroup.java:963)
at android.view.ViewGroup.unFocus(ViewGroup.java:963)
at android.view.ViewGroup.unFocus(ViewGroup.java:963)
at android.view.ViewGroup.unFocus(ViewGroup.java:963)
at android.view.ViewGroup.unFocus(ViewGroup.java:963)
at android.view.ViewGroup.removeViewInternal(ViewGroup.java:4149)
at android.view.ViewGroup.removeViewInternal(ViewGroup.java:4137)
at android.view.ViewGroup.removeView(ViewGroup.java:4069)
at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1529)
at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1759)
at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1827)
at android.support.v4.app.FragmentManagerImpl.dispatchStateChange(FragmentManager.java:3244)
at android.support.v4.app.FragmentManagerImpl.dispatchDestroy(FragmentManager.java:3235)
at android.support.v4.app.FragmentController.dispatchDestroy(FragmentController.java:265)
at android.support.v4.app.FragmentActivity.onDestroy(FragmentActivity.java:390)
at android.support.v7.app.AppCompatActivity.onDestroy(AppCompatActivity.java:209)
at android.app.Activity.performDestroy(Activity.java:6112)
at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1140)
```

**Screenshots of the issue**

> Not available, crash screen

**Would you like to work on the issue?**

@Vikum94

--- a/app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java
+++ b/app/src/main/java/org/fossasia/pslab/fragment/LuxMeterFragmentConfig.java
@@ -88,20 +88,24 @@ public class LuxMeterFragmentConfig extends Fragment {
         highLimit.setOnFocusChangeListener(new View.OnFocusChangeListener() {
             @Override
             public void onFocusChange(View v, boolean hasFocus) {
-                if (!hasFocus) {
-                    String stringValue = highLimit.getText().toString();
-                    int value = 0;
-                    try {
-                        value = Integer.parseInt(stringValue);
-                    } catch (NumberFormatException e) {
-                        e.printStackTrace();
+                try {
+                    if (!hasFocus) {
+                        String stringValue = highLimit.getText().toString();
+                        int value = 0;
+                        try {
+                            value = Integer.parseInt(stringValue);
+                        } catch (NumberFormatException e) {
+                            e.printStackTrace();
+                        }
+                        if (value > highLimitMax)
+                            highLimitSeek.setProgress(highLimitMax);
+                        else if (value < 0)
+                            highLimitSeek.setProgress(0);
+                        else
+                            highLimitSeek.setProgress(value);
                     }
-                    if (value > highLimitMax)
-                        highLimitSeek.setProgress(highLimitMax);
-                    else if (value < 0)
-                        highLimitSeek.setProgress(0);
-                    else
-                        highLimitSeek.setProgress(value);
+                } catch (NullPointerException e) {
+                    e.printStackTrace();
                 }
             }
         });
@@ -127,20 +131,24 @@ public class LuxMeterFragmentConfig extends Fragment {
         updatePeriod.setOnFocusChangeListener(new View.OnFocusChangeListener() {
             @Override
             public void onFocusChange(View v, boolean hasFocus) {
-                if (!hasFocus) {
-                    String stringValue = updatePeriod.getText().toString();
-                    int value = 100;
-                    try {
-                        value = Integer.parseInt(stringValue);
-                    } catch (NumberFormatException e) {
-                        e.printStackTrace();
+                try {
+                    if (!hasFocus) {
+                        String stringValue = updatePeriod.getText().toString();
+                        int value = 100;
+                        try {
+                            value = Integer.parseInt(stringValue);
+                        } catch (NumberFormatException e) {
+                            e.printStackTrace();
+                        }
+                        if (value > updatePeriodMax)
+                            updatePeriodSeek.setProgress(updatePeriodMax + 100);
+                        else if (value < updatePeriodMin)
+                            updatePeriodSeek.setProgress(updatePeriodMin - 100);
+                        else
+                            updatePeriodSeek.setProgress(value - updatePeriodMin);
                     }
-                    if (value > updatePeriodMax)
-                        updatePeriodSeek.setProgress(updatePeriodMax + 100);
-                    else if (value < updatePeriodMin)
-                        updatePeriodSeek.setProgress(updatePeriodMin - 100);
-                    else
-                        updatePeriodSeek.setProgress(value - updatePeriodMin);
+                } catch (NullPointerException e) {
+                    e.printStackTrace();
                 }
             }
         });

